Feat/replays for prague with besu and issue1335

amkCha opened the pull request:
When running a replay test for an lt trace + shomei trace

use `besuNodeFastReplayTests` here
<img width="834" height="691" alt="image" src="https://github.com/user-attachments/assets/c4881da8-1cd1-4fdc-b004-8cbb7d8012e3" />

Comment line 99-106 in `net/consensys/linea/testing/ExecutionEnvironment.java` to keep the traces

and you can find traces in 
<img width="525" height="137" alt="image" src="https://github.com/user-attachments/assets/e2620679-5366-4ccf-bd90-36af325cc161" />

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Increase timeouts and block-build delay in Besu replay tools and update `ConflationSnapshot.from` usage to include fork name.
> 
> - **Testing**:
>   - **BesuExecutionTools**:
>     - Set `OkHttpClient` read timeout to 30s.
>     - Increase block build delay: `blockBuildingTimeMs` from `period*1000` to `period*10000`.
>     - Extend trace file wait timeout from `10` to `60` seconds.
>   - **MultiBlockExecutionEnvironment**:
>     - Update `ConflationSnapshot.from(...)` call to include `testsChain.fork.name()` (new API signature).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 6f562a257684df6868839705c0fe8b0d76cc8d23. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

amkCha committed:
feat add replay for issue 1335

amkCha committed:
feat: change timeouts to run replays

amkCha commented on arithmetization/src/main/java/net/consensys/linea/blockcapture/snapshots/BlockHeaderSnapshot.java:
```diff
@@ -87,14 +87,10 @@ public BlockHeader toBlockHeader() {
             .prevRandao(Bytes32.fromHexString(this.mixHashOrPrevRandao))
             .nonce(this.nonce)
-            .blockHeaderFunctions(new CliqueBlockHeaderFunctions());
+            .blockHeaderFunctions(new MainnetBlockHeaderFunctions())
```
I'm not sure if it's the right class to use from now on - to be checked with @fab-10 :pray:

amkCha replied:
but it is working though :)

DavePearce replied:
The problem you could have here is that old replay tests may fail, as they require Clique.  However, that perhaps depends on whether `MainnetblockHeaderFunctions` accounts for this or not.

amkCha commented on testing/src/main/java/net/consensys/linea/testing/BesuExecutionTools.java:
```diff
@@ -97,7 +98,7 @@ public BesuExecutionTools(
     this.testName = tmpTestName.substring(0, Math.min(tmpTestName.length(), 200));
     int besuPort = findFreePort();
     int shomeiPort = findFreePort();
```
In this class, we change all the timeouts to allow bigger blocks and tracers to be traced than our regular units are doing

amkCha committed:
feat: rename resource file

DavePearce committed:
fix missing SPDX header

DavePearce committed:
revert parentBeaconBlockRoot to Optional<String>

This resets the parentBeaconBlockRoot implementation back to what it was
originally, as this is required for two reasons.  Firstly, to support
forks which don't have a parent beacon block root (i.e. those prior to
CANCUN).  Secondly, to support older replays which don't even have an
entry for this field.

DavePearce committed:
embed fork name into replay files

This updates the BlockCapturer so that it embeds the fork name (e.g.
"PRAGUE") into captured replays.  This is not strictly required, but
seems like useful metadata for debugging.  Observe that the fork is not
used within the tracer, it doesn't matter whether it is actually in the
replay or not.

DavePearce committed:
tweak setting for coinbase address

The coinbase address is either determined from the block header, or from
the blocks "extra data".  The latter is done when the Clique consensus
protocol is being used (i.e. all forks prior to PRAGUE).  This fix
simply ensures that old replays will *not* take the coinbase address
from the block header.

DavePearce reviewed (approved):
LGTM!
